clippy_lints/src/functions/renamed_function_params.rs

```diff
@@ -0,0 +1,123 @@
+use clippy_utils::def_path_def_ids;
+use clippy_utils::diagnostics::span_lint_and_then;
+use rustc_errors::{Applicability, MultiSpan};
+use rustc_hir::def_id::{DefId, DefIdSet};
+use rustc_hir::hir_id::OwnerId;
+use rustc_hir::{Impl, ImplItem, ImplItemKind, ImplItemRef, ItemKind, Node, TraitRef};
+use rustc_lint::LateContext;
+use rustc_span::symbol::{kw, Ident, Symbol};
+use rustc_span::Span;
+use std::sync::OnceLock;
+
+use super::RENAMED_FUNCTION_PARAMS;
+
+static IGNORED_TRAIT_IDS: OnceLock<DefIdSet> = OnceLock::new();
```

Comment:
I'm not sure if I should use `OnceLock` here, haven't seen them in other lints, maybe there's some kind of concern? idk

Reply:
I believe some other lints store this in the lint pass struct instead. The IDs can then be collected in `fn check_crate()`. I would prefer that solution over having a static object. Otherwise, I don't believe there is a reason against `OnceLock`

Reply:
Oh I see, I just thought it might be redundant to having a list of trait paths, and a set of their ids stored in the same struct